dwave/system/composites/cutoffcomposite.py

```diff
@@ -0,0 +1,364 @@
+# Copyright 2019 D-Wave Systems Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+# ================================================================================================
+"""
+A composite that removes all variables smaller than the provided cutoff
+
+"""
+
+from heapq import heapify, heappop
+
+import numpy as np
+from dimod import BinaryQuadraticModel, SampleSet
+from dimod.core.composite import ComposedSampler
+from dimod.higherorder import _relabeled_poly, poly_energies
```

Comment:
Should not import private functions, either need to make that function public or recreate it here.

I am not sure we should be supporting HUBOs in lots of composites, can become confusing for users who may expect others to support them as well. A more elegent solution might be to implement a new solver ABC that requires the existence of a `sample_hubo` method. We should discuss.

Reply:
See https://github.com/dwavesystems/dimod/issues/391